### PR TITLE
refactor(sd): Remove selenium

### PR DESCRIPTION
**Pull request by flooie**
Update SD and backscraper

Update tests for new non selenium world.

this is by far my worst refactor.
the least confident.

### CODE REVIEW

File: juriscraper/opinions/united_states/state/sd.py [hunk 1]
```diff
@@ -3,106 +3,40 @@
...
-from juriscraper.lib.string_utils import convert_date_string, titlecase
-from juriscraper.OpinionSiteWebDriven import OpinionSiteWebDriven
 
-
-class Site(OpinionSiteWebDriven):
+class Site(OpinionSiteLinear):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.court_id = self.__module__
+        self.status = "Published"
```
Comment (mlissner) [inline juriscraper/opinions/united_states/state/sd.py:19]: Hm, is this being used somewhere? I think this attribute isn't ever used?

File: juriscraper/opinions/united_states/state/sd.py [hunk 2]
```diff
@@ -3,106 +3,40 @@
...
+            self.cases.append(
+                {
+                    "date": get_row_column_text(row, 1),
+                    "name": titlecase(
+                        get_row_column_text(row, 2).rsplit(",", 1)[0]
+                    ),
+                    "neutral_citation": cite[0],
+                    "url": row.xpath(".//td[2]/a/@href")[0],
+                    "docket": row.xpath(".//td[2]/a/@href")[0].split("/")[-1][
+                        :5
```
Comment (mlissner) [inline juriscraper/opinions/united_states/state/sd.py:39]: Would love some examples of what the text is here.

**Review by mlissner — CHANGES_REQUESTED** [review]
Hm, this looks pretty good to me, despite you saying you're uncertain about it.

File: juriscraper/opinions/united_states/state/sd.py (hunk 1 above)
Comment (flooie) [reply juriscraper/opinions/united_states/state/sd.py:19]: yes - status is required on all scrapers.  This just sets status to published for the SD supremes

File: juriscraper/opinions/united_states/state/sd.py (hunk 2 above)
Comment (flooie) [reply juriscraper/opinions/united_states/state/sd.py:39]: this may be a second candidate for extracting from the PDF.

Comment (flooie) [reply juriscraper/opinions/united_states/state/sd.py:39]: we extract the start of the docket.  but we dont extract the full docket.

Comment (mlissner) [reply juriscraper/opinions/united_states/state/sd.py]: Sounds good. I'll hang tight for that, then.

Comment (flooie) [reply juriscraper/opinions/united_states/state/sd.py]: this has been added.

File: juriscraper/opinions/united_states/state/sd.py [hunk 3]
```diff
@@ -3,106 +3,59 @@
...
-    def extract_regex_group(self, group, string, alt=False):
-        regex = self.regex_alt if alt else self.regex
-        return re.search(regex, string, re.MULTILINE).group(group)
+    def extract_from_text(self, scraped_text: str) -> Dict[str, Any]:
+        """Can we extract the date filed from the text?
+
+        :param scraped_text: The content of the document downloaded
+        :return: Metadata to be added to the case
+        """
+        docket = re.findall(r"#\d+.*-.-\w{3}", scraped_text)[0][:20]
```
Comment (mlissner) [inline juriscraper/opinions/united_states/state/sd.py:55]: Sorry, I keep mentioning this, but when I see `[:20]`, I worry. Can you put a comment about what's going on here with an example of what's getting parsed?

File: juriscraper/opinions/united_states_backscrapers/state/sd.py [hunk 4]
```diff
@@ -1,229 +1,60 @@
...
+        self.parameters = {
+            "ctl00$ctl00$ScriptManager1": "ctl00$ctl00$ContentPlaceHolder1$ChildContent1$UpdatePanel_Opinions|ctl00$ctl00$ContentPlaceHolder1$ChildContent1$LinkButton_Next",
+            "__EVENTTARGET": "ctl00$ctl00$ContentPlaceHolder1$ChildContent1$LinkButton_Next",
+        }
+        self.parameters["__VIEWSTATE"] = (
+            self.html.xpath("//*[@id='__VIEWSTATE']/@value")[0],
+        )
+        self.parameters["__EVENTVALIDATION"] = (
+            self.html.xpath("//*[@id='__EVENTVALIDATION']/@value")[0],
+        )
```
Comment (mlissner) [inline juriscraper/opinions/united_states_backscrapers/state/sd.py:28]: I missed this before, but let's get the xpath stuff out of the set parameters method, please.

**Review by mlissner — CHANGES_REQUESTED** [review]

File: juriscraper/opinions/united_states/state/sd.py (hunk 3 above)
Comment (flooie) [reply juriscraper/opinions/united_states/state/sd.py:59]: sounds good

File: juriscraper/opinions/united_states_backscrapers/state/sd.py (hunk 4 above)
Comment (flooie) [reply juriscraper/opinions/united_states_backscrapers/state/sd.py]: √

**Comment by flooie** [comment]
Updated the Backscrapers to include a better yearly parsing

**Comment by flooie** [comment]
@mlissner I think this is also ready to be merged.

**Comment by mlissner** [comment]
> @mlissner I think this is also ready to be merged.

Sounds good. Merging without further review.

**Review by mlissner — APPROVED** [review]

**Pull request merged**